src/features/game/Humans.tsx

```diff
@@ -10,18 +11,29 @@ import mapMovement from "./lib/mapMovement";
 import { ExpansionInfo } from "./expansion/ExpansionInfo";
 import { useParams } from "react-router-dom";
 import { Section, useScrollIntoView } from "lib/utils/hooks/useScrollIntoView";
+import { useActor } from "@xstate/react";
 
 export const Humans: React.FC = () => {
+  const { authService } = useContext(Auth.Context);
+  const [authState, send] = useActor(authService);
   // catching and passing scroll container to keyboard listeners
   const container = useRef(null);
   const { id } = useParams();
   const [scrollIntoView] = useScrollIntoView();
 
+  useEffect(() => {
+    // Move auth machine out of visiting state if route changed via browser back button
+    if (
+      authState.matches({ connected: "visitingContributor" }) ||
```

Comment:
How can a player be in the `visiting` or `visitingContributor` state if they are in the `Humans` component?

I find putting logic in useEffect a bit disconnected at times. It is not clear what logic was fired to get into this state. It is a bit confusing how this logic would get called.

Is there somewhere else we could put this business logic and fire `send("RETURN");`?

Reply:
This is confusing, I agree but its because of the way our machines/navigation are set up.

When a user is authorised and playing the game they can visit a contributor via the town hall. When they do this their `gameMachine` dies and they spin up a `visitingMachine. This is controlled by the navigation ie `/visit...` loads `Visiting` which starts machine. Through an event we also move the `authMachine` into the `{ connected: 'visitingContributor' }` state since this machine is outside of the navigation. 
This makes for a better experience when a user hits in the `Back` button because they can just go back to their game rather than having to go through the entire connecting flow again which is unnecessary as we already have all of their details sitting in the authMachine. We have control over this because we control the back button with events. 
However, if a user uses the browser back button we are unable to fire an event to the authMachine because, like I mentioned above, it exists outside of our navigation. To get around this, inside of the `Humans` component which is loaded on the game route, on load, we check the current state of the authMachine and if it is in one of the visiting states then move it back into its correct state by sending the `RETURN` event. If we didn't do this then it would be stuck in a visiting state when a user is back playing the game therefore we would miss events.

We can definitely fix this we just need to think about a scaleable solution to our machine/navigation architecture. It's too big to include in this PR obviously. I have added a clearer comment.